### PR TITLE
Add module crm_lead_invoice_address.

### PR DESCRIPTION
**Pull request by yajo**
Relaunching part of Antiun/antiun-odoo-addons#11.

Add invoice address fields to leads.

It will fully work after odoo/odoo#8899 is merged, but I think it can be merged anyway.

cc @rafaelbn.

### CODE REVIEW

**Comment by yajo** [comment]
Now NOT depending on odoo/odoo#8899.

**Comment by rafaelbn** [comment]
Tested in runbot :+1:

**Comment by pedrobaeza** [comment]
Please check Travis

**Comment by yajo** [comment]
Travis seems unrelated. Same as https://github.com/OCA/crm/pull/52#issuecomment-148747755.

File: crm_lead_invoice_address/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingeniería, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Invoice address in leads",
+    "summary": "Add Invoice address field to leads",
+    "version": "8.0.1.0.0",
+    "category": "Customer Relationship Management",
+    "invoice address": "https://odoo-community.org/",
```
Comment (osvalr) [inline crm_lead_invoice_address/__openerp__.py]: Is this a valid manifest key? it is used?

**Comment by janetrp** [comment]
can't move on with test because the error Object is not callable remains:

```
 File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm_lead_supplier/models/crm_lead.py", line 21, in _lead_create_contact
    ._lead_create_contact(lead, name, is_company, parent_id))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
TypeError: 'NoneType' object is not callable
```

**Comment by yajo** [comment]
See https://github.com/OCA/crm/pull/55#issuecomment-150581678.

Please remind to put logs inside triple-backticks to make them easier to read:

``````
```
[the logs]
```
``````

 :wink:

**Comment by yajo** [comment]
Travis is fixed, please check again. :smile:

File: crm_lead_invoice_address/__openerp__.py (hunk 1 above)
Comment (rafaelbn) [inline crm_lead_invoice_address/__openerp__.py]: Not sure

**Comment by rafaelbn** [comment]
Tested
- Create a lead write adrress and invoice address:
- If you don't fill street 2:

```
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118393-54-bcb263/openerp/api.py", line 345, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118393-54-bcb263/openerp/addons/crm_lead_invoice_address/models.py", line 34, in _lead_create_contact
    "street2": lead.invoice_street2 or result.get("street2"),
AttributeError: 'int' object has no attribute 'get'
```
- If you fill street 2 but not fill ZIP

```
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118393-54-bcb263/openerp/api.py", line 345, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118393-54-bcb263/openerp/addons/crm_lead_invoice_address/models.py", line 35, in _lead_create_contact
    "zip": lead.invoice_zip or result.get("zip"),
AttributeError: 'int' object has no attribute 'get'
```

This errors comes when any field is nor filled.

File: crm_lead_address_street3/model/crm_lead.py [hunk 2]
```diff
@@ -21,31 +21,21 @@
...
+
+    @api.multi
+    def on_change_partner_id(self, partner_id):
+        res = super(Lead, self).on_change_partner_id(partner_id)
         if partner_id:
-            partner = self.pool['res.partner'].browse(cr, uid, partner_id,
-                                                      context=context)
+            partner = self.env['res.partner'].browse(partner_id)
             res['value'].update({'street3': partner.street3})
         return res
```
Comment (carlosdauden) [inline crm_lead_address_street3/model/crm_lead.py:41]: This file belongs to this module?
This Is merged in #61

**Comment by yajo** [comment]
@rafaelbn Actually you found quite a tricky bug, I'm on it.

**Comment by yajo** [comment]
Latest commits fix https://github.com/OCA/crm/pull/54#issuecomment-151855745 and tes it. Now it is required to have both partner and contact names to assign invoice address to one and normal address to another. Also there was a bug where both addresses could get mixed which is tested and fixed too.

**Comment by yajo** [comment]
Last commits fix https://github.com/OCA/crm/pull/53#issuecomment-152221135, affecting this one too.

**Comment by janetrp** [comment]
tested in runbot
Converted the lead in opportunity 
Customer get address 1
Contact get address 2
:+1:

**Comment by carlosdauden** [comment]
:+1:

**Comment by rafaelbn** [comment]
Tested :+1:

**Pull request merged**